### PR TITLE
misc: Add Pannotia GPU tests

### PR DESCRIPTION
**Pull request by mkjost0**
This adds the Pannotia GPU tests to be run weekly, though currently the binaries for these tests aren't compiling properly due to being unable to find m5ops, so they are still a work in progress.

Change-Id: I7feb28aa1921e67ecacd084e84edcf43a87241ba

### CODE REVIEW

**Review by mattsinc — COMMENTED** [review]
Since this is a draft commit and the m5ops aren't working, I'm assuming you don't want an actual review yet -- let me know.

**Comment by abmerop** [comment]
Is there something different about this application vs. other applications using m5ops that is causing the compilation issues?

**Comment by mattsinc** [comment]
@mkjost0 : now that https://github.com/gem5/gem5-resources/pull/5 and https://github.com/gem5/gem5/pull/186 have been merged in, minimally lines 74-76 will need to change to use the new input args for FW.  @jebraun3 please help @mkjost0 with what change is needed.

Maximally, we should work together to figure out how to update the YAML file to the new format James used in https://github.com/gem5/gem5/pull/186 in weekly.sh -- it looks like 74-76 don't use that format.  We are happy to help in any way here to make it easy so James' subsequent patches can update the weekly YAML file.

**Comment by jebraun3** [comment]
@mkjost0 Sorry for responding to this so late. For FW to work, the options in line 76 would need to change to be `--options="-f 1k_128k.gr -m default"`. To fully utilize the new changes, FW should be run on the underlying hardware (not in gem5) with the options `--options="-f 1k_128k.gr -m generate"`, then in gem5 with the options `--options="-m usemmap"`. Please let me know if you have any questions.

File: .github/workflows/gpu-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        with:
+          name: ${{ needs.name-artifacts.outputs.build-name }}${{ matrix.image }}
+          path: build/${{ matrix.image }}/gem5.opt
+          retention-days: 5
+      - run: echo "This job's status is ${{ job.status }}."
+
+# These tests fail to make gem5-fusion due to not finding gem5/m5ops.h
+  pannotia-tests:
+    runs-on: [self-hosted, linux, x64, build]
+    container: mkjost/testing-gpu-image:latest
```
Comment (BobbyRBruce) [inline .github/workflows/gpu-tests.yaml:56]: @mkjost0 : Do you have the Dockerfile for this image? I don't know what this image consists of and I don't appear to have permissions to pull it

**Comment by BobbyRBruce** [comment]
Closing as this appears to have abandoned. An issue has been created to track the adding of Pannotia to the GPU tests if desired: #1422

**Comment by mattsinc** [comment]
> Closing as this appears to have abandoned. An issue has been created to track the adding of Pannotia to the GPU tests if desired: #1422

I was not aware the GPU tests had been removed from weekly ... do you know what they were failing on?

**Comment by BobbyRBruce** [comment]
I

> > Closing as this appears to have abandoned. An issue has been created to track the adding of Pannotia to the GPU tests if desired: #1422
> 
> I was not aware the GPU tests had been removed from weekly ... do you know what they were failing on?

I don't have a good grasp of this, only to say that this was a change Melissa had yet to finish upon leaving. YOu can see I tried back in November to take over her work but this PR is dependent on a docker image she didn't provide sources for. I'd probably advise anyone taking over this task to just start from scratch otherwise it's confusing

I also was a little surprised we never got these in. I felt we had, but I checked and they were not translated from the jenkins to the GitHub Actions runner system. They were just here for someone else to pick up.

**Pull request closed without merging**